core/shared/src/test/scala/shapeless/conversions.scala

```diff
@@ -64,28 +64,40 @@ class ConversionTests {
   def testFunctions {
     val sum : (Int, Int) => Int = _+_
     val prd : (Int, Int, Int) => Int = _*_*_
-    
+
     val hlsum = sum.toProduct
     typed[(Int :: Int :: HNil) => Int](hlsum)
-    
+
     val hlprd = prd.toProduct
     typed[(Int :: Int :: Int :: HNil) => Int](hlprd)
-    
+
     trait A
     trait B extends A
     trait C extends A
-    
+
     val a = new A {}
     val b = new B {}
-    
+
     val ab : A => B = (a : A) => b
-    
+
     val hlab = ab.toProduct
     typed[(A :: HNil) => B](hlab)
- 
+
     def foo[F, L <: HList, R](f : F, l : L)(implicit fntp: FnToProduct.Aux[F, L => R]) = fntp(f)(l)
     val s2 = foo(sum, 2 :: 3 :: HNil)
     val ab2 = foo(ab, a :: HNil)
+
+    class HListSyntax[A <: HList, F <: AnyRef](a: A) {
+       def $$[U](f: F)(implicit cftp: FnToProduct.Aux[f.type, A => U]): U = cftp(f)(a)
```

Comment:
I'm not completely convinced by `$$` as a symbolic operator for this. People who don't know the Haskell won't get it, and people who do will get it back to front. Maybe provide a non symbolic name and try `$:` (is that legal?) as the symbolic variant?